src/cryptography/x509.py

```diff
@@ -669,6 +681,29 @@ class SubjectKeyIdentifier(object):
     def __init__(self, digest):
         self._digest = digest
 
+    @classmethod
+    def from_public_key(cls, public_key):
+        # This is a very slow way to do this.
+        serialized = public_key.public_bytes(
+            serialization.Encoding.DER,
+            serialization.PublicFormat.SubjectPublicKeyInfo
+        )
+        spki, remaining = decoder.decode(
+            serialized, asn1Spec=_SubjectPublicKeyInfo()
+        )
+        # the univ.BitString object is a tuple of bits. We need bytes and
+        # pyasn1 really doesn't want to give them to us. To get it we'll
+        # build an integer, hex it, then decode the hex.
+        bits = 0
+        for bit in spki.getComponentByName("subjectPublicKey"):
+            bits = bits << 1 | bit
+
+        # convert the integer to bytes
+        hex_string = '%x' % bits
```

Comment:
We can switch to hex() but then we'll need to [2:] to strip the 0x and .rstrip("L") to get rid of the L in Python 2. That seems uglier, but admittedly `%` is a weird beast in Python with its "not recommended but never going to be removed" status.

If you'd prefer it to swap to `hex(bits)[2:].rstrip("L")` I'll change it.